src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -230,7 +230,17 @@ def dependencies(
         if entry is None:
             raise self._coordinate_not_found(key, coord)
 
-        return (entry, tuple(entries[(i.group, i.artifact)] for i in entry.dependencies))
+        return (
+            entry,
+            tuple(
+                dependency_entry
+                for d in entry.dependencies
+                # If the dependency is missing from the entries, we want to skip the dependency.
```

Comment:
Maybe something like:

```suggestion
                # If the dependency is missing from the entries, we want to skip the dependency because Coursier has decided it's not needed.
```